### ISSUE TITLE
Event Registration Block Errors Out When Adding Mobile Phone to Use Default Value on Registration Form

### ISSUE THREAD

**Issue by austinspooner** [SparkDevNetwork/Rock#5619]
### Description

When "Mobile Phone" is added to an Event Registration Template as a Registrant Form Field and "Use Default Value" is selected, an exception error is generated on the Registration page when a user trys to visit the page after registering (such as via the Registration List Lava block or the Sent Payment Remdinder communication. 

<img width="1174" alt="image" src="https://github.com/SparkDevNetwork/Rock/assets/657005/1e0ad2e9-31ae-4882-8e7a-d74d280d7b94">

### Actual Behavior

Actual Behavior
An exception error is generated.

### Expected Behavior

Expected Behavior

User would be able to visit the page and make a payment successfully. 

### Steps to Reproduce

See this video:
https://app.screencast.com/pt9HoZHnk5fg0

![image](https://github.com/SparkDevNetwork/Rock/assets/657005/c0b04a20-ebdc-402f-8baf-007d035683d0)

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

16

### Client Culture Setting

en-US

**Comment by dataCollegechurch** [comment]
Just including some extra details here in case it helps someone locate this bug a bit easier

The exception message associated with this issue is `Object reference not set to an instance of an object.`

My exception occurred at `Rock.Blocks.Event.RegistrationEntry.GetPersonCurrentFieldValue(RockContext rockContext, Person person, RegistrationTemplateFormField field, RegistrationContext registrationContext) in \Rock.Blocks\Event\RegistrationEntry.cs:line 1503`

**Comment by nairdo** [comment]
@dataCollegechurch  I'm confused by your comment above.  That is reported to have been fixed in v16.1.  Are you saying you are still seeing this issue in v16.1?

**Comment by dataCollegechurch** [comment]
@nairdo - no I experienced the issue while on 16.0. Originally when I did a search for it on github I did not find it. I am happy anytime someone takes the time to submit a bug, but the original details of the issue were a bit sparse i.e. "An exception error is generated." I thought it may be helpful if I included some basic details about the exception message to improve the searchability of this issue.
